hilti/toolchain/src/base/util.cc

```diff
@@ -199,21 +173,25 @@ hilti::Result<hilti::rt::filesystem::path> util::findInPaths(const hilti::rt::fi
     return hilti::result::Error(fmt("%s not found", file));
 }
 
-hilti::rt::filesystem::path util::currentExecutable() {
-    const auto exe = PathFind::FindExecutable();
-
-    if ( exe.empty() ) {
-        auto msg = std::string("could not determine path of current executable");
-
-#if defined(__FreeBSD__)
-        if ( ! rt::filesystem::exists("/proc") || rt::filesystem::is_empty("/proc") )
-            msg += ": /proc needs to be mounted";
-#endif
-
-        rt::internalError(msg);
+hilti::rt::filesystem::path util::currentExecutable(const hilti::rt::filesystem::path& argv0) {
+    // We only use the error code for calls to `exists` to prevent any
+    // exceptions. If that function returns `true` the file exists and one can
+    // ignore the error code.
+    std::error_code ec;
+
+    // If `argv0` exists the executable was invoked directly.
```

Comment:
I don't think this is true:  if argv0 is `exe` and there's `exe` inside the current path and also in `PATH`, this would return the former, but we're probably running the latter. 

I don't fully recall but I believe this ambiguity is reason for using pathfinder in the first place. Not quite sure what the right way out is without it.